File: mp/src/main/java/ro/ubb/mp/dao/model/Appointment.java
package ro.ubb.mp.dao.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity(name = "appointments")
public class Appointment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    private User student;

    @ManyToOne(fetch = FetchType.LAZY)
    private User mentor;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:[.ss][.SSS][.SS][.S]")
    private LocalDateTime date;

    private String locationDetails;
}

File: fe/src/types/Appointment.ts
import { Dayjs } from 'dayjs'

export interface UserAppointment {
  id: number
  fullName: string
}

export interface Appointment {
  id: number
  student: UserAppointment
  mentor: UserAppointment
  date: Dayjs
  locationDetails: string
}

export interface AppointmentResponseDto {
  id: number
  student: UserAppointment
  mentor: UserAppointment
  date: string
  locationDetails: string
}

export interface AppointmentRequestDto {
  id: number
  student: UserAppointment
  mentor: UserAppointment
  date: string
  locationDetails: string
}
